pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -154,6 +153,27 @@ func (esc *endpointSliceController) onEventInternal(_, ep *v1.EndpointSlice, eve
 	}
 }
 
+func serviceNeedsPush(svc *corev1.Service) bool {
+	if svc == nil {
+		return false
+	}
+	if svc.Spec.ClusterIP != corev1.ClusterIPNone || svc.Spec.Type == corev1.ServiceTypeExternalName {
```

Comment:
hmm I don't get this, why would we skip a push if its a normal service?

Reply:
This was a mistake. Fixed now